pkg/kubelet/kubelet.go

```diff
@@ -375,6 +374,7 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 	crOptions *config.ContainerRuntimeOptions,
 	containerRuntime string,
 	hostnameOverride string,
+	nodeName types.NodeName,
```

Comment:
I'm curious how/why this duplication arose - are there calls to `NewMainKubelet` that aren't in `cmd/...` which could be relying on the internal calculation of the `nodeName`?

Reply:
Just confirmed. `NewMainKubelet` is only used in `cmd/kubelet/server.go`

https://github.com/kubernetes/kubernetes/blob/master/cmd/kubelet/app/server.go#L1178

Reply:
Kubemark calls directly into `RunKubelet`, so that would skip the initial hostname calculation IIRC.

Reply:
Right, hollow_kubelet.go ---> RunKubelet (calculation IIRC)-> createAndInitKubelet-> NewMainKubelet (calculation IIRC again)

--
It seems that NewMainKubelet does the calculation again IIRC is not necessary